### PR TITLE
feat: Included 1.x rules for aws cis 1.4.0

### PR DESCRIPTION
**Pull request by m-pizarro**

### CODE REVIEW

File: src/aws/cis-1.4.0/rules/aws-cis-1.4.0-1.19.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      accountId
+      __typename
+      expiration
+    }
+  }`,
+  resource: 'queryawsIamServerCertificate[*]',
+  severity: 'high',
+  conditions: {
+    value: { daysDiff: {}, path: '@.expiration' },
+    greaterThanInclusive: 1,
```
Comment (tyler-dunkel) [inline src/aws/cis-1.4.0/rules/aws-cis-1.4.0-1.19.ts:66]: so what does this condition say? "a cert will pass so long as its expiration is at least 1 day in the future"?

File: src/aws/cis-1.4.0/rules/aws-cis-1.4.0-1.7.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      arn
+      accountId
+       __typename
+      passwordLastUsed
+      passwordEnabled
+    }
+  }`,
+  resource: 'queryawsIamUser[*]',
+  severity: 'high',
+  conditions: {
```
Comment (tyler-dunkel) [inline src/aws/cis-1.4.0/rules/aws-cis-1.4.0-1.7.ts:56]: reading the description, do we also need to check the access key last used here?

Comment (tyler-dunkel) [reply src/aws/cis-1.4.0/rules/aws-cis-1.4.0-1.7.ts:56]: or check that it has no access keys

**Comment by tyler-dunkel** [comment]
To ensure consistency, I think we need a `:` between the rule prefix (aws CIS 1.4) and the actual rule title. Also please remove any `(manual)` strings from the title. We can add those in programatically.

File: src/aws/cis-1.4.0/rules/aws-cis-1.4.0-1.19.ts (hunk 1 above)
Comment (m-pizarro) [reply src/aws/cis-1.4.0/rules/aws-cis-1.4.0-1.19.ts:66]: Yes, daysDiff returns the days between now and a later date. So this checks that the expiration date is greater than today's date.

**Comment by tyler-dunkel** [comment]
> > 
> 
> Ok, I'll remove all "(manual)" legends. As for the` :` in the title, you says something like this "AWS CIS 1.7`: `Eliminate use of the root user for administrative and daily tasks", if so, this was not added for any policy package (except for pci)

Hmm I see,  I its fine to do it with no `:` so long as its consistent but check with Ale as it may be easier for EP if the `:` is there.

File: src/aws/cis-1.4.0/rules/aws-cis-1.4.0-1.7.ts (hunk 2 above)
Comment (tyler-dunkel) [reply src/aws/cis-1.4.0/rules/aws-cis-1.4.0-1.7.ts:60]: This part of the sql query looks to check against access keys, we should probably do the same (check access keys and password)
```
when access_key_1_last_used_date <= (current_date - interval '90' day)  then 'alarm'
    when access_key_2_last_used_date <= (current_date - interval '90' day)  then 'alarm'
```

Comment (m-pizarro) [reply src/aws/cis-1.4.0/rules/aws-cis-1.4.0-1.7.ts:60]: @tyler-dunkel  yes, I saw that too and it is correct. But also that is what is being done in the rule, we are validating the password and both access keys. We return both (key_1 and key_2) inside the accessKeyData array. Unless I'm missing something.

Comment (tyler-dunkel) [reply src/aws/cis-1.4.0/rules/aws-cis-1.4.0-1.7.ts:60]: I might be missing something. I only see the password enabled part of the rule conditions here

Comment (tyler-dunkel) [reply src/aws/cis-1.4.0/rules/aws-cis-1.4.0-1.7.ts:60]: nvm I see the changes now

**Review by tyler-dunkel — APPROVED** [review]

**Pull request merged**